docs(TLS): Adds more detail about using listener certificates (#7675)
strimzi/strimzi-kafka-operator: documentation/modules/security/proc-installing-certs-per-listener.adoc

// assembly-security.adoc

[id='proc-installing-certs-per-listener-{context}']

> = Providing your own Kafka listener certificates for TLS authentication

Reviewer: ```suggestion
= Providing your own Kafka listener certificates for TLS encryption
```

[role="_abstract"]
Listeners provide client access to Kafka brokers.

> You configure listeners in the `Kafka` resource, including the configuration required for client access using TLS.

Reviewer 2: ```suggestion
Configure listeners in the `Kafka` resource, including the configuration required for using TLS with client access.
```

> You configure listeners in the `Kafka` resource, including the configuration required for client access using TLS authentication.

Reviewer: I would not mention the authentication here => that will again create confusion that the custom certificate is used for client authentication.

```suggestion
You configure listeners in the `Kafka` resource, including the configuration required for client access using TLS.
```

> By default, the listeners use certificates signed by the internal CAs generated by Strimzi.

Reviewer 2: Since these modules can be standalone, I think you should introduce the expanded wording for CAs first

> By default, the listeners use certificates signed by the internal self-signed CAs generated by Strimzi.

Reviewer: Strictly speaking, every root CA is self-signed. So I would leave it out here.

```suggestion
By default, the listeners use certificates signed by the internal CAs generated by Strimzi.
```

> A public CA certificate is generated by the Cluster Operator when it creates a Kafka cluster.

Reviewer: Public CA would normally be understood as a well recognized CA such as LEt's encrypt, which is not the case here.

```suggestion
A CA certificate is generated by the Cluster Operator when it creates a Kafka cluster.
```

> When you configure a client for TLS, you add the public CA certificate to its truststore configuration to verify the Kafka cluster.

Reviewer 3: as suggested before let's avoid "public" here

You can also xref:installing-your-own-ca-certificates-str[install and use your own CA certificates]. 
Or you can configure a listener using `brokerCertChainAndKey` properties and use a custom server certificate.

> The `brokerCertChainAndKey` properties allow you to authenticate access to Kafka brokers using your own custom certificates at the listener-level.

Reviewer: Again, I would avoid the `authentication` word here.

You create a secret with your own private key and server certificate, then specify the key and certificate in the listener's `brokerCertChainAndKey` configuration.

> The server certificate must be signed by a CA.

Reviewer: It does not have to be signed by a CA. But if it would not be, there is no sense in using the custom certificate. So I would remove this.

> If signed by a public CA, you won't usually need to add it to a client's truststore configuration.  

Reviewer 2: ```suggestion
If signed by a public CA, you usually won't need to add it to a client's truststore configuration.  
```

> You add the public CA that signed the server certificate to the client's truststore configuration.

Reviewer: Again, public CA would normally be something like Let's Encrypt => a publicly recognized "public" CA. You do not have to use a server cert signed by a public CA. It can be private CA as well. And in case it is public CA such as Let's Encrypt, you normally don't need to add it to a truststore.

Author: Added this to explain the situation:

```
You can use a certificate signed by a public (external) CA or a private CA.
If signed by a public CA, you won't usually need to add it to a client's truststore configuration. 
```

Custom certificates are not managed by Strimzi, so you need to renew them manually. 

NOTE: Listener certificates are used for server authentication. 

> For client authentication, you add `authentication.type: TLS` to your listener and Kafka user configuration for the client.

Reviewer 3: `tls` lowercase

> NOTE: Listener certificates are used for server authentication. 
> For client authentication, you add `authentication.type: TLS` to your listener and Kafka user configuration for the client.
> You must also add a public user certificate to the keystore configuration of the client.
> The public user certificate must be signed by the clients CA certificate generated by the Cluster Operator.
> Or you can xref:installing-your-own-ca-certificates-str[install and use your own clients CA]. 

Reviewer: Too much detail which I think leads away from the main message

```suggestion
NOTE: Listener certificates are used for TLS encryption and server authentication only. 
It is not used for TLS client authentication.
If you want to use your own certificate for TLs client authentication as well, you have to xref:installing-your-own-ca-certificates-str[install and use your own clients CA]. 
```

* The Cluster Operator is running.
* Each listener requires the following:
** A compatible server certificate signed by an external CA. (Provide an X.509 certificate in PEM format.)
+
You can use one listener certificate for multiple listeners.

> ** Subject Alternative Names (SANs) specified in the certificate for each listener.

Reviewer 2: ```suggestion
** Subject Alternative Names (SANs) are specified in the certificate for each listener.
```

> ** Subject Alternative Names (SANs) specified in each certificate for each listener.

Reviewer: I think that quite often the users have one certificate for multiple listeners. Maybe this can be incorporated here? Right, I think it might be read by some as if each listener needs to have a different certificate.

Author: Changed to:

> * Each listener requires the following:
> ** A compatible server certificate signed by an external CA. (Provide an X.509 certificate in PEM format.)
> +
> You can use one listener certificate for multiple listeners.
> ** Subject Alternative Names (SANs) specified in the certificate for each listener.
> For more information, see xref:ref-alternative-subjects-certs-for-listeners-{context}[].
